src/commonMain/kotlin/exchange.dydx.abacus/state/manager/notification/NotificationsProvider.kt

```diff
@@ -0,0 +1,74 @@
+package exchange.dydx.abacus.state.manager.notification
+
+import exchange.dydx.abacus.output.Asset
+import exchange.dydx.abacus.output.Notification
+import exchange.dydx.abacus.protocols.ParserProtocol
+import exchange.dydx.abacus.state.manager.V4Environment
+import exchange.dydx.abacus.state.manager.notification.providers.BlockRewardNotificationProvider
+import exchange.dydx.abacus.state.manager.notification.providers.OrderStatusChangesNotificationProvider
+import exchange.dydx.abacus.state.model.TradingStateMachine
+import exchange.dydx.abacus.utils.IMap
+import exchange.dydx.abacus.utils.JsonEncoder
+import exchange.dydx.abacus.utils.ParsingHelper
+import exchange.dydx.abacus.utils.UIImplementations
+import kollections.toIMap
+
+interface NotificationsProviderProtocol {
+    fun buildNotifications(
+        stateMachine: TradingStateMachine,
+        subaccountNumber: Int
+    ): IMap<String, Notification>
+
+    fun asset(stateMachine: TradingStateMachine, marketId: String): Asset? {
```

Comment:
Why is this method here? This seems like a fairly generic util method that could be placed somewhere else?

Reply:
This only used by the child providers, so probably no need to expose it outside.

Reply:
I don't think adding a method to an interface is the right way to go about this. Interfaces should only expose methods for consumers, not helper functions for implementations.

Reply:
OK.. Updated